webapp/src/pages/UserStatsPage/UserStatsPage.tsx

```diff
@@ -17,44 +16,107 @@ import 'chartjs-adapter-date-fns';
 import { Header, IconVariant } from '../../shared/components';
 import { useNavigation } from '../../shared/hooks/UseNavigation';
 
+type FiveArray<T> = [T, T, T, T, T];
+type proArray = [
+  UserLevelWithTimestamp[],
+  GameControllerGetUserLevelHistoryModeEnum,
+];
 export default function UserStatsPage() {
   const { username } = useParams()!;
   const { goBack } = useNavigation();
-  const getLevelHistory = useCallback(
-    () =>
-      gameApi.gameControllerGetUserLevelHistory({
-        username: username!,
-        mode: GameControllerGetUserLevelHistoryModeEnum.Classic,
-      }),
+  const getUserByUserName = useCallback(
+    () => usersApi.userControllerGetUserByUserName({ userName: username! }),
     [username],
   );
-  const { data: levels, isLoading: areLevelsLoading } =
-    useData<UserLevelWithTimestamp[]>(getLevelHistory);
+  const { data: user, isLoading: isUserLoading } = useData(getUserByUserName);
+  const getLevelHistory = useCallback(async (): Promise<
+    FiveArray<UserLevelWithTimestamp[]>
+  > => {
+    try {
+      if (!user) {
+        return Promise.reject('Unresolved');
+      }
+      let pclassic: proArray = [
```

Comment:
There's a better way to do it: populate all arrays with the fake entry first, and THEN use the callbacks.
But I was too exhausted last night to be able to see it.

Reply:
will do.

Reply:
suggestion: We could fetch all the user levels and convert them to an object where the key is the game mode, and the value is the levels for that game mode. This way, we make one fetch and process it on the client.

<details>

```tsx
type GameModeLevels = {
  [key in GameControllerGetUserLevelHistoryModeEnum]: UserLevelWithTimestamp[];
};

...

  const getGameModeLevels = (): GameModeLevels => {
    let initialGameModeLevels: GameModeLevels = {
      training: [],
      classic: [],
      shortPaddle: [],
      mysteryZone: [],
      unknown: [],
    };
    if (!levels) {
      return initialGameModeLevels;
    }
    const gameModesWithLevels = levels.reduce<GameModeLevels>(
      (gameModeLevels, level) => {
        switch (level.gameMode) {
          case 'training':
            return {
              ...gameModeLevels,
              training: [...gameModeLevels.training, level],
            };
          case 'classic':
            return {
              ...gameModeLevels,
              classic: [...gameModeLevels.classic, level],
            };
          case 'mysteryZone':
            return {
              ...gameModeLevels,
              mysteryZone: [...gameModeLevels.mysteryZone, level],
            };
          case 'shortPaddle':
            return {
              ...gameModeLevels,
              shortPaddle: [...gameModeLevels.shortPaddle, level],
            };
          case 'unknown':
            return {
              ...gameModeLevels,
              unknown: [...gameModeLevels.unknown, level],
            };
          default:
            return gameModeLevels;
        }
      },
      initialGameModeLevels,
    );
    return gameModesWithLevels;
  };

...

          <Line
            data={{
              datasets: Object.entries(getGameModeLevels()).map((entry) => {
                const [gameMode, gameModeLevels] = entry;
                return {
                  label: gameMode,
                  data: gameModeLevels.map((item) => ({
                    x: item.timestamp,
                    y: item.level,
                  })),
                  borderWidth: 1,
                  stepped: true,
                };
              }),
            }}
            ... />

```

</details>

Reply:
We'll have to change the endpoint as well. As of now, without a mode, it returns only the ones for classic mode. Will do.

Reply:
Also, I'll still need to add a fake entry in the beginning, when the user registers. Any other solution to that?

Reply:
> We'll have to change the endpoint as well. As of now, without a mode, it returns only the ones for classic mode. Will do.

You are right :+1:

> Also, I'll still need to add a fake level 1 entry in the beginning of all game modes, when the user registers. Any other solution to that?

I think we don't need to add the fake entry, or am I missing something?

Reply:
Well, unlike a XP oriented system, in an elo system every user, when coming to existence, is assigned a level. I do think we need it, if only to provide a real progression of the level, and to make it pretty. I cannot insert it in the database because the user levels are referenced to games, also I wouldn't want to. It's a matter of representation, which is webapp responsibility.

Bottom line: I absolutely intend to add it.

Reply:
done